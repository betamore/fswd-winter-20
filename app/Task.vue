<template>
    <li>
        <router-link :to="'/tasks/' + task.id"> {{ task.name }}</router-link>
        <span v-if="taskIsCompleted">is completed</span>
    </li>
</template>

<script>
export default {
    props: ["task"],
    computed: {
        taskIsCompleted() {
            return this.task.name.length % 2 === 0;
        }
    }
};
</script>
