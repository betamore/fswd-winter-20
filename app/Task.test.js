/**
 * @jest-environment jsdom
 */

import { shallowMount, mount } from "@vue/test-utils";
import Task from "./Task.vue";

describe("Task vue component", function() {
    it("should render something", function() {
        const wrapper = shallowMount(Task, {
            propsData: { task: { id: 123, name: "Task Name" } },
            stubs: ["router-link", "router-view"]
        });
        expect(wrapper.text()).toBe("Task Name");
    });

    it("should show is completed", function() {
        const wrapper = shallowMount(Task, {
            propsData: { task: { id: 123, name: "Task Names" } },
            stubs: ["router-link", "router-view"]
        });
        expect(wrapper.text()).toBe("Task Names is completed");
    });

    it("should match the snapshot", function() {
        const wrapper = mount(Task, {
            propsData: { task: { id: 123, name: "Task Names" } },
            stubs: ["router-link", "router-view"]
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});
