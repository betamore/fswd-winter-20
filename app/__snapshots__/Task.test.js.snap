// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task vue component should match the snapshot 1`] = `
<li>
  <router-link-stub
    to="/tasks/123"
  >
     Task Names
  </router-link-stub>
   
  <span>
    is completed
  </span>
</li>
`;
